src/fides/api/util/tcf/tc_mobile_data.py

```diff
@@ -0,0 +1,56 @@
+from fides.api.schemas.privacy_experience import TCMobileData
+from fides.api.util.tcf.tc_model import TCModel
+from fides.api.util.tcf.tc_string import (
+    PURPOSE_CONSENTS_BITS,
+    PURPOSE_LEGITIMATE_INTERESTS_BITS,
+    SPECIAL_FEATURE_BITS,
+    USE_NON_STANDARD_TEXT_BITS,
+    TCField,
+    _get_max_vendor_id,
+    build_tc_string,
+    get_bits_for_section,
+)
+
+
+def build_tc_data_for_mobile(tc_model: TCModel) -> TCMobileData:
+    """Build TC Data for Mobile App"""
+
+    def _build_binary_string(name: str, num_bits: int) -> str:
+        """Internal helper to build a bit string of 0's and 1's to represent list data
+        pulled from the TC model using the prescribed number of bits"""
+        return get_bits_for_section([TCField(name=name, bits=num_bits)], tc_model)
+
+    # If vendors have consent or legitimate interest data, show the max id that exists.
+    # This will end up being the bitstring length of these sections
+    max_vendor_consents: int = _get_max_vendor_id(tc_model.vendor_consents)
+    max_vendor_li: int = _get_max_vendor_id(tc_model.vendor_legitimate_interests)
+
+    tc_string: str = build_tc_string(tc_model)
+
+    return TCMobileData(
```

Comment:
Most of the content of this model is just translating the internal TCModel into the requested format for display -